scripts/checksha.py

```diff
@@ -0,0 +1,139 @@
+    parser.add_argument(
+        '-s', '--sha', type=str, default=None,
+        help='provide a parent sha rather than interrogating the wheels '
+             'that are deployed in the solution to get the sha')
+
+    return parser.parse_args()
+
+
+def find_inventory(inv, path):
+    '''Find the dynamic inventory file'''
+    for root, dirs, files in os.walk(path):
+        if inv in files:
+            return os.path.join(root, inv)
+
+
+def clone_git_repo(project, path=CLONE_DIR):
+    '''clone a git repo'''
+    os.chdir(path)
+    if not os.path.exists(project):
+        print 'cloning %s project to interrogate the history...' \
+              'please be patient this may take a while' % project
+        try:
+            subprocess.check_call(
+                'git clone https://github.com/openstack/%s.git '
+                '--quiet' % project,
+                shell=True, stderr=subprocess.STDOUT)
+        except CalledProcessError as e:
+            print 'Something went wrong when trying to clone the repo'
+            sys.exit(e)
+
+
+def parse_results(results, project):
+    '''parse and print results of the ansible run'''
+    shas = set()
+    hostnames = []
+    for (hostname, result) in results['contacted'].items():
+        shas.add(result['stdout'].split()[3])
+        hostnames.append(hostname)
+    # bail if multiple or no shas found
+    if len(shas) == 0:
+        print 'no hosts are running project %s...exiting' % project
+        sys.exit(1)
+    elif len(shas) > 1:
+        print 'you seem to have multiple versions of the %s wheel ' \
+              ' deployed in your solution...exiting'
+        sys.exit(1)
+    else:
+        sha = list(shas)[0]
+        print 'contacted these hosts:'
+        print '======================'
+        print '\n'.join(sorted(hostnames))
+        print '\nThe %s wheel was built from the following sha' % project
+        print '==============================================='
+        print sha + '\n'
+        return sha
+
+
+def is_ancestor(commit, sha):
+    '''check if commit is an ancestor of sha'''
+    # (NOTE) git merge-base returns no output, just a return code:
+    # 1: not an ancestor
+    # 0: is an ancestor
+    # 128: not a valid git object (for either sha or commit)
+    # other: OS error
+    try:
+        subprocess.check_call(
+            'git merge-base --is-ancestor %s %s' % (commit, sha), shell=True)
+        return True
+    except CalledProcessError as e:
+        if e.returncode == 1:
+            # not an ancestor
+            return False
+        else:
+            # command went wrong
+            sys.exit(e)
+
+
+def main(args):
+    project = args.project
+    commit = args.commit
+
+    # run pbr info on {project} on all hosts that might have it
+    dyn_inventory = find_inventory(INVENTORY_FILENAME, BASE_PATH)
+    inventory = ansible.inventory.Inventory(dyn_inventory)
+    results = ansible.runner.Runner(
+        pattern='%s_all' % project,
```

Comment:
This seems a bit limiting (that it only works for openstack 'type:service' projects). If this is all that's required by support? I think the project help string in get_args should be more explicit if that is the case.

Alternatively, why to allow an ansible group and wheel to be supplied. That gives support the ability to check any wheel.